cooked-validators/src/Cooked/Attack/AddToken.hs

```diff
@@ -0,0 +1,61 @@
+module Cooked.Attack.AddToken where
+
+import Control.Monad
+import Cooked.Attack.Tweak
+import Cooked.MockChain.Wallet
+import Cooked.Tx.Constraints.Type
+import qualified Data.Map as Map
+import qualified Ledger as Pl
+import qualified Ledger.Value as Pl
+import qualified Ledger.Value as Value
+import qualified PlutusTx.Numeric as Pl
+import Test.QuickCheck.Modifiers (NonZero)
+
+-- | This attack adds extra tokens, depending on the minting policy. It is
+-- different from the 'dupTokenAttack' in that it does not merely try to
+-- increase the amount of tokens minted: It tries to mint tokens of asset
+-- classes that were not necessarily present on the unmodified transaction.
+--
+-- This attack adds an 'AddTokenLbl' with the token name of the additional
+-- minted token(s). It returns additional value minted.
+addTokenAttack ::
+  -- | For each policy that occurs in some 'Mints' constraint, return a list of
+  -- token names together with how many tokens with that name should be
+  -- minted.
+  --
+  -- For each of the elements of the returned list, one modified transaction
+  -- with the additional tokens will be generated. (This means for example that,
+  -- if there were three minting policies on the original transaction, and the
+  -- lists returned for each of them have n,m, and o elements, respectively,
+  -- there'll be n*m*o modified transactions.)
+  --
+  -- The redeemer will be the one that's already being used on the transaction.
+  (Pl.Versioned Pl.MintingPolicy -> [(Pl.TokenName, NonZero Integer)]) ->
+  -- | The wallet of the attacker. Any extra tokens will be paid to this wallet.
+  Wallet ->
+  Tweak Pl.Value
+addTokenAttack extraTokens attacker = do
+  oldMints <- viewTweak txSkelMintsL
+  msum $
+    map
+      ( \policy ->
+          let (redeemer, _) = oldMints Map.! policy
```

Comment:
Would it be better to do `Map.toList` instead of `Map.keys` to avoid this lookup (and the cognitive overhead of convincing oneself it's always successful etc)?